Lib/asyncio/__main__.py

```diff
@@ -97,30 +97,16 @@ def run(self):
                     exec(startup_code, console.locals)
 
             ps1 = getattr(sys, "ps1", ">>> ")
-            if can_colorize():
+            if can_colorize() and CAN_USE_PYREPL:
                 ps1 = f"{ANSIColors.BOLD_MAGENTA}{ps1}{ANSIColors.RESET}"
             console.write(f"{ps1}import asyncio\n")
 
-            try:
-                import errno
-                if os.getenv("PYTHON_BASIC_REPL"):
-                    raise RuntimeError("user environment requested basic REPL")
-                if not os.isatty(sys.stdin.fileno()):
-                    return_code = errno.ENOTTY
-                    raise OSError(return_code, "tty required", "stdin")
-
-                # This import will fail on operating systems with no termios.
```

Comment:
I have a suspicion that the failing CI is caused by removing this.

Reply:
Nah, those checks now happen in `_pyrepl.main`. The actual failure is explained in the last commit's message.